casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -347,18 +348,20 @@ object Validate {
     } yield status
   }
 
-  // Validator should only process deploys from its own shard
+  // Validator should only process deploys from its own shard with shard names in ASCII characters only
   def deploysShardIdentifier[F[_]: Monad: Log](
       b: BlockMessage,
       shardId: String
-  ): F[ValidBlockProcessing] =
+  ): F[ValidBlockProcessing] = {
+    assert(shardId.onlyAscii, "Shard name should contain only ASCII characters")
```

Comment:
This will crash the node if block with non ascii shard is received?
I think this should not be assert but a condition next to `b.body.deploys.forall(_.deploy.data.shardId == shardId)`

Reply:
Oh, I see - node cannot start with non ASCII shard and wrong shard blocks will be dropped on receive. Is this the logic assumed?